src/App.jsx

```diff
@@ -18,17 +18,28 @@ import AboutPage from './components/presentations/AboutPage/AboutPage';
 import Profilepage from './components/containers/profile.container';
 import ForgotPassword from './components/containers/passwordReset.containers';
 import ResetPassword from './components/presentations/PasswordReset/ResetPassword';
+import SignupContainer from './components/containers/signup.container';
 import { decodeToken } from './utils/authService';
 
 class App extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = { user: null };
   }
 
   componentDidMount() {
     const user = decodeToken();
     this.setState({ user });
+    if (user) {
+      if (!user.isActivated) {
```

Comment:
If we can have this logic in one if statement let's do that.

Reply:
Nice will work on that